onboarding/source/amb_light_service.c

```diff
@@ -57,34 +56,30 @@ uint8_t initLightService(void) {
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
-    adcData_t adc_data;
-    adcData_t *adc_data_ptr = &adc_data;
+    adcData_t adcData;
+    light_event_t *currentEvent = NULL;
 
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
     while (1) {
-        if (xQueueReceive(xQueue, NULL, (TickType_t) 11) == pdPASS) {
-            adcStartConversion(adcREG1, 
-                                adcGROUP1);
+        if (xQueueReceive(xQueue, currentEvent, portMAX_DELAY) == pdPASS) {
+            
+            if(currentEvent == MEASURE_LIGHT) {
```

Comment:
`currentEvent` is a pointer, so you can't compare it to `MEASURE_LIGHT`. You should dereference the pointer.

However, you shouldn't have a separate pointer variable anyways because of what I mentioned in my other comment.